lib/praxis/types/multipart_array.rb

```diff
@@ -150,6 +148,67 @@ def self.example(context=Attributor::DEFAULT_ROOT_CONTEXT, **options)
         example
       end
 
+
+      def self.describe(shallow=true, example: nil)
+        type_name = Attributor.type_name(self)
+        hash = {
+          name: type_name.gsub(Attributor::MODULE_PREFIX_REGEX, ''),
+          family: self.family,
+          id: self.id
+        }
+        hash[:example] = example if example
+
+        hash[:part_name] = {type: name_type.describe(true)}
+
+        unless shallow
+          hash[:attributes] = {} if self.attributes.keys.any? { |name| name.kind_of? String}
+          hash[:pattern_attributes] = {} if self.attributes.keys.any? { |name| name.kind_of? Regexp}
+
+          if hash.key?(:attributes) || hash.key?(:pattern_attributes)
+            self.describe_attributes(shallow, example: example).each do |name, sub_hash|
+              if self.multiple.include?(name)
+                #sub_hash[:options] ||= {}
+                #sub_hash[:multiple] = true
+              end
+
+              case name
+              when String
+                hash[:attributes][name] = sub_hash
+              when Regexp
+                hash[:pattern_attributes][name.source] = sub_hash
+              end
+
+            end
+          else
+            hash[:part_payload] = {type: payload_type.describe(true)}
+          end
+        end
+        hash
+      end
+
+      def self.describe_attributes(shallow=true, example: nil)
+        self.attributes.each_with_object({}) do |(part_name, part_attribute), parts|
+          sub_hash = part_attribute.describe(shallow, example: example)
```

Comment:
hmmm...here, we should pass only the sub-example to the describe...not the whole thing.
...it'd be interesting to see how we pass the sub-example for patterns...I guess we can if we actually apply the match...